cmd/kueuectl/app/create/create_clusterqueue.go

```diff
@@ -0,0 +1,489 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	kueuev1beta1 "sigs.k8s.io/kueue/client-go/clientset/versioned/typed/kueue/v1beta1"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+	utilslices "sigs.k8s.io/kueue/pkg/util/slices"
+)
+
+const (
+	cqLong    = `Creates a ClusterQueue with the given name.`
+	cqExample = `  # Create a ClusterQueue 
+  kueuectl create clusterqueue my-cluster-queue
+  
+  # Create a ClusterQueue with cohort, namespace selector and other details
+  kueuectl create clusterqueue my-cluster-queue \
+  	--cohort=cohortname \
+	--queuing-strategy=StrictFIFO \
+	--namespace-selector=fooX=barX,fooY=barY \
+	--reclaim-within-cohort=Any \
+	--preemption-within-cluster-queue=LowerPriority
+  
+  # Create a ClusterQueue with nominal quota and one resource flavor named alpha
+  kueuectl create clusterqueue my-cluster-queue --nominal-quota=alpha:cpu=0;memory=0
+  
+  # Create a ClusterQueue multiple resource flavors named alpha, beta and gamma
+  kueuectl create clusterqueue my-cluster-queue \
+  	--nominal-quota=alpha:cpu=0;memory=0,beta:gpu=0,gamma:cpu=0;memory=0 \
+	--borrowing-limit=alpha:cpu=0;memory=0,beta:gpu=0,gamma:cpu=0;memory=0 \
+	--lending-limit=alpha:cpu=0;memory=0,beta:gpu=0,gamma:cpu=0;memory=0`
```

Comment:
Use nvidia.com/gpu instead of gpu. Additoonaly, resource flavor with just gpu doesn't make sense - any workload using it will need a bit of cpu and mem.